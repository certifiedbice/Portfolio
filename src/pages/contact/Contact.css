/*---------- # Contact ---------------------------------------*/
.contact-text-group{
	display: flex;
	flex-direction: column;
	flex-wrap: wrap;
}
.contact-text{
	flex: 3 calc(33.3%);
	margin: 10px;
	background-color: transparent;
	box-shadow: none;
	padding: 20px 0 0 0;
	vertical-align: middle;
	text-align: center;
	font-family: '3_thehard_way_rmxfenotype';
	font-size: 66px;
	line-height:2;
	color: #fff;
	text-shadow: 2px 2px 4px #000;
}
/* Desktop CSS */
@media(min-width:700px){
.contact-text{
	padding: 20px 0 0 0;
	font-size: 42px;
}
}
#contact-button{
	color: #666;
	cursor: pointer;
}
#contact-button:hover{
	color: #222;
	text-shadow: 2px 2px 4px #fff;
}
#contact-social-icon-links-container{
	background-color: transparent;
	box-shadow: none;
	text-align: center;
}
.contact-social-icon-link{
  	cursor: pointer;
}
.contact-social-icon-image{
	max-height: 500px;
	max-width: 500px;
}
/* Desktop CSS */
@media(min-width:700px){
.contact-social-icon-image{
	max-height: 100px;
	max-width: 100px;
}
}
#contact-overlay{
	display: none;
	position: fixed;
	top: 0;
	left: 0;
	height: 100%;
	width: 100%;
	background-color: #000;
	background: rgba(0, 0, 0, 0.5);
	z-index: 99;
	opacity: 0.0;
}
#contact-form-container{
	position: relative;
	flex: 1;
	align-self: center;
	margin: auto;
	max-width: 90%;
	background-color: #fff;
	background: rgba(255,255,255, 1);
	border-radius: 10px;
	padding: 10px;
	overflow: hidden;
}
#contact-form-elements-container{
	display: flex;
	flex-direction: column;
	max-width: 99%;
}
#contact-form-elements-container *{
	flex: 1;
	align-self: center;
	width: 98%;
	border: 1px solid #000;
	border-radius: 5px;
	padding: 5px;
	font-size: 10px;
}
/* Desktop CSS */
@media(min-width:700px){
#contact-form-elements-container *{
	width: 100%;
}
}
#contact-form-elements-container form{
  	border: none;
}
#contact-form-elements-container legend{
	border: none;
	padding-left: 25px;
	text-align: center;
	line-height: 2;
	font-family: '3_thehard_way_rmxfenotype';
	font-size: 18px;
	font-weight: bold;
	color: #000;
}
#contact-form-elements-container label{
  	border: none;
}
#contact-form-elements-container input, textarea{
	margin-bottom: 10px;
	box-shadow: inset 0px 0px 5px #333;
	-moz-box-shadow: inset 0px 0px 5px #333;
	-webkit-box-shadow: inset 0px 0px 5px #333;
}
#contact-form-submit-button{
	cursor: pointer;
}
#contact-form-status{
	position: relative;
	/* display: none; */
	display: flex;
	justify-content: center;
	/* flex:1; */
	/* flex-direction: column; */
	align-self: center;
	margin-top: 20px;
	max-width: 99%;
	padding: 5px;
	font-family: '3_thehard_way_rmxfenotype';
	font-size: 20px;
	line-height:2;
	text-align: center;
	color: #0f0;
}
#contactHideBtn{
	position: absolute;
	top: 9px;
	right: 10px;
	height: 15px;
	width: 26px;
	border: 1px solid #000;
	border-radius: 10px;
	padding: 7px 0 5px 0;
	text-align: center;
	color: #000;
	cursor: pointer;
}
#contactHideBtn:hover{
	background-color: #000;
	color: #fff;
}