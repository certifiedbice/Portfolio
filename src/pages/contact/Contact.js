import React,{Component} from 'react';
import Context from '../../masterProvider';
import DataHelper from '../../dataHelper';
import Header from '../../components/header/Header';
import Footer from '../../components/footer/Footer';
import Fade from '../../components/fade/Fade';
import Background from '../../components/background/Background';
import './Contact.css';
export default class Contact extends Component{
	static contextType=Context;
	showContactForm=()=>{
		this.context.contactFade(true);
	}
	hideContactForm=()=>{
		this.context.contactFade(false);
	}
	componentDidMount(){
		// Compensate for content outside the viewport.
		this.context.verticalPadAndScroll();
		// Blink the email link.
		this.context.emailLinkBlink();
		// FormSpree
		// helper function for sending an AJAX request
		function ajax(method, url, data, success, error){
			var xhr = new XMLHttpRequest();
			xhr.open(method, url);
			xhr.setRequestHeader("Accept", "application/json");
			xhr.onreadystatechange = function() {
				if (xhr.readyState !== XMLHttpRequest.DONE) return;
				if (xhr.status === 200) {
					success(xhr.response, xhr.responseType);
				} else {
					error(xhr.status, xhr.response, xhr.responseType);
				}
			};
			xhr.send(data);
		}
		const closeContactForm=()=>{this.context.contactFade(false);}
		window.addEventListener("DOMContentLoaded", function() {
			// get the form elements defined in your form HTML above
			var form = document.getElementById("contact-form");
			const el1=document.querySelector('#contact-form-status');
			// Success and Error functions for after the form is submitted
			function success() {
				form.reset();
				el1.innerHTML='Submission Success';
				setTimeout(closeContactForm(),3000)
			}
			function error() {
				el1.innerHTML = 'Oops! There was a problem.';
			}
			// handle the form submission event
			form.addEventListener("submit", function(ev) {
				ev.preventDefault();
				var data = new FormData(form);
				ajax(form.method, form.action, data, success, error);
			});
		});
	}
	componentWillUnmount(){
		this.context.emailLinkBlink(true);
	}
	render(){
		const socialInfo=DataHelper.social;
		return(
			<>
				<Background/>
				<main>
					<section id='contact'>
						<div className='group contact-text-group'>
							<div id='contact-button' className='item contact-text'onClick={this.showContactForm}>Email</div>
							<div className='item contact-text'>or</div>
							<div className='item contact-text'>Social</div>
						</div>
						<div className='group'>
							<div id='contact-social-icon-links-container' className="item">
								{
									socialInfo.map(item=>{
										if(item.link!==null){
											return (
												<a key={item.title} href={item.link} className='contact-social-icon-link' target='_blank' rel='noopener noreferrer'>
													<img className='contact-social-icon-image' src={require(`../../images/social-icons/${item.icon}`)} alt={item.title}/>
												</a>
											);
										}
										else return null;
									})
								}
							</div>
						</div>
						<div id='contact-overlay'>
							<div id='contact-form-container'>
								<div id='contact-form-elements-container'>
									<div id="contactHideBtn" onClick={this.hideContactForm}>X</div>
									<form id='contact-form' action='https://formspree.io/xnqgpoje' method='POST'>
										<legend>Email Me</legend>
										<label htmlFor='fname'>First name:</label>
										<input type='text' id='fname' placeholder='John'/>
										<label htmlFor='lname'>Last name:</label>
										<input type='text' id='lname' placeholder='Doe'/>
										<label htmlFor='email'>Email Address:</label>
										<input type='text' id='email' placeholder='john.doe@eod.nhoj.com'/>
										<label htmlFor='phone'>Phone Number:</label>
										<input id='phone' type='text' name='phone' pattern='\d{3}[\-]\d{3}[\-]\d{4}' placeholder='xxx-xxx-xxxx'/>
										<label htmlFor='organization'>Organization:</label>
										<input type='text' id='organization' placeholder='John Doe Corporation'/>
										<label htmlFor='subject'>Subject:</label>
										<input type='text' id='subject' placeholder='I am lost, do you know who I am?'/>
										<label htmlFor='comments'>Comments:</label>
										<textarea name='comments' id='comments' cols='30' rows='8' maxLength='200'></textarea>
										<input id='contact-form-submit-button' type='submit' value='Submit'/>
									</form>
								</div>
								<div id='contact-form-status'></div>
							</div>
						</div>
					</section>
				</main>
				<Header/>
				<Footer/>
				<Fade location={this.props.location.pathname}/>
			</>
		);
	}
}