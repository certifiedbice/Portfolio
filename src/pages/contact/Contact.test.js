import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Contact from './Contact';

it('renders without crashing',()=>{
	const pathname={location:{pathname:'/contact'}};
    const div=document.createElement('div');
    ReactDOM.render(<BrowserRouter><Contact location={pathname}/></BrowserRouter>,div);
});

it('renders correctly',()=>{
	const pathname={location:{pathname:'/contact'}};
    const tree=renderer
        .create(<BrowserRouter><Contact location={pathname}/></BrowserRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});