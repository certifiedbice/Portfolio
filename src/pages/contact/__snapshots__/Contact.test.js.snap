// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    id="background-image-container"
  >
    <div
      id="background-image"
    />
  </div>,
  <main>
    <section
      id="contact"
    >
      <div
        className="group contact-text-group"
      >
        <div
          className="item contact-text"
          id="contact-button"
          onClick={[Function]}
        >
          Email
        </div>
        <div
          className="item contact-text"
        >
          or
        </div>
        <div
          className="item contact-text"
        >
          Social
        </div>
      </div>
      <div
        className="group"
      >
        <div
          className="item"
          id="contact-social-icon-links-container"
        >
          <a
            className="contact-social-icon-link"
            href="https://www.facebook.com/certifiedbice"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Facebook"
              className="contact-social-icon-image"
              src="facebook.png"
            />
          </a>
          <a
            className="contact-social-icon-link"
            href="https://github.com/certifiedbice"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="GitHub"
              className="contact-social-icon-image"
              src="github.png"
            />
          </a>
          <a
            className="contact-social-icon-link"
            href="https://www.instagram.com/certifiedbice"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Instagram"
              className="contact-social-icon-image"
              src="instagram.png"
            />
          </a>
          <a
            className="contact-social-icon-link"
            href="https://www.hackerrank.com/certifiedbice"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="HackerRank"
              className="contact-social-icon-image"
              src="hackerrank.png"
            />
          </a>
          <a
            className="contact-social-icon-link"
            href="https://www.linkedin.com/in/certifiedbice"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="LinkedIn"
              className="contact-social-icon-image"
              src="linkedin.png"
            />
          </a>
          <a
            className="contact-social-icon-link"
            href="https://twitter.com/certifiedbice"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Twitter"
              className="contact-social-icon-image"
              src="twitter.png"
            />
          </a>
          <a
            className="contact-social-icon-link"
            href="https://www.youtube.com/channel/UCEITFHDFDtvTIKFbUhG_E9w"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="YouTube"
              className="contact-social-icon-image"
              src="youtube.png"
            />
          </a>
        </div>
      </div>
      <div
        id="contact-overlay"
      >
        <div
          id="contact-form-container"
        >
          <div
            id="contact-form-elements-container"
          >
            <div
              id="contactHideBtn"
              onClick={[Function]}
            >
              X
            </div>
            <form
              action="https://formspree.io/xnqgpoje"
              id="contact-form"
              method="POST"
            >
              <legend>
                Email Me
              </legend>
              <label
                htmlFor="fname"
              >
                First name:
              </label>
              <input
                id="fname"
                placeholder="John"
                type="text"
              />
              <label
                htmlFor="lname"
              >
                Last name:
              </label>
              <input
                id="lname"
                placeholder="Doe"
                type="text"
              />
              <label
                htmlFor="email"
              >
                Email Address:
              </label>
              <input
                id="email"
                placeholder="john.doe@eod.nhoj.com"
                type="text"
              />
              <label
                htmlFor="phone"
              >
                Phone Number:
              </label>
              <input
                id="phone"
                name="phone"
                pattern="\\\\d{3}[\\\\-]\\\\d{3}[\\\\-]\\\\d{4}"
                placeholder="xxx-xxx-xxxx"
                type="text"
              />
              <label
                htmlFor="organization"
              >
                Organization:
              </label>
              <input
                id="organization"
                placeholder="John Doe Corporation"
                type="text"
              />
              <label
                htmlFor="subject"
              >
                Subject:
              </label>
              <input
                id="subject"
                placeholder="I am lost, do you know who I am?"
                type="text"
              />
              <label
                htmlFor="comments"
              >
                Comments:
              </label>
              <textarea
                cols="30"
                id="comments"
                maxLength="200"
                name="comments"
                rows="8"
              />
              <input
                id="contact-form-submit-button"
                type="submit"
                value="Submit"
              />
            </form>
          </div>
          <div
            id="contact-form-status"
          />
        </div>
      </div>
    </section>
  </main>,
  <header
    id="site-header"
  >
    <div
      id="site-info"
    >
      <h1
        id="site-title"
      >
        Portfolio of Bice
      </h1>
      <div
        id="site-description"
      >
        My various projects in all their variousness.
      </div>
      <nav>
        <ul
          id="primary-menu"
        >
          <li>
            <a
              href="/home"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="/portfolio"
            >
              Portfolio
            </a>
          </li>
          <li>
            <a
              href="/resume"
            >
              Resume
            </a>
          </li>
          <li>
            <a
              href="/services"
            >
              Services
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>,
  <footer
    id="site-footer"
  >
    Bice - 2020
    <div
      id="footer-links-container"
    >
      <a
        className="footer-social-icon-link"
        href="https://www.facebook.com/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Facebook"
          className="footer-social-icons-image"
          src="facebook.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://github.com/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="GitHub"
          className="footer-social-icons-image"
          src="github.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://www.instagram.com/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Instagram"
          className="footer-social-icons-image"
          src="instagram.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://www.hackerrank.com/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="HackerRank"
          className="footer-social-icons-image"
          src="hackerrank.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://www.linkedin.com/in/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="LinkedIn"
          className="footer-social-icons-image"
          src="linkedin.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://twitter.com/certifiedbice"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Twitter"
          className="footer-social-icons-image"
          src="twitter.png"
        />
      </a>
      <a
        className="footer-social-icon-link"
        href="https://www.youtube.com/channel/UCEITFHDFDtvTIKFbUhG_E9w"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="YouTube"
          className="footer-social-icons-image"
          src="youtube.png"
        />
      </a>
    </div>
  </footer>,
  <div
    id="page-fade"
  />,
]
`;
